sdk/examples/wallet/participation.rs

```diff
@@ -1,206 +1,256 @@
+//!
+//! Rename `.env.example` to `.env` first, then run the command:
+//! ```sh
+//! cargo run --release --all-features --example wallet_participation
+//! ```
 
-use std::str::FromStr;
+use std::{env::var, str::FromStr};
 
 use iota_sdk::{
-    client::{
-        constants::SHIMMER_COIN_TYPE,
-        node_manager::node::Node,
-        request_funds_from_faucet,
-        secret::{mnemonic::MnemonicSecretManager, SecretManager},
-    },
+    client::node_manager::node::Node,
     types::api::plugins::participation::types::ParticipationEventId,
-    wallet::{account::types::participation::ParticipationEventRegistrationOptions, ClientOptions, Result, Wallet},
+    wallet::{account::types::participation::ParticipationEventRegistrationOptions, Result, Wallet},
     Url,
 };
 
+// The node that runs the participation plugin
+const PARTICPATION_NODE_URL: &str = "https://api.testnet.shimmer.network";
+// Some (voted on) participation event. You need to change it to perform an actual vote, otherwise the example will
+// abort at the time where it tries to send a voting transaction.
+const PARTICIPATION_EVENT_ID_1: &str = "0x5a5d145648cd5c100e64d4463f5cccf994a404dcc58e2c3bdfef3aa82266aa8d";
+// An ignored participation event. You can empty the string if you don't care about it!
+const IGNORED_PARTICIPATION_EVENT_ID: &str = "0x8f682b31fb9d9ff57d87dd6061d823a355eafe133f5d40f96aaca5c5a3d6fc5d";
+// A deregistered participation event. You can empty the string if you don't care about it!
+const DEREGISTERED_PARTICIPATION_EVENT: &str = "0x16f4b8fa61f40a666404ac446b6a74a0dea47342345311294676794d0dc8b67a";
+// The amount of voting power we'll increase
+const INCREASE_VOTING_POWER_AMOUNT: u64 = 1000001;
+// The amount of voting power we'll then decrease
+const DECREASE_VOTING_POWER_AMOUNT: u64 = 1;
+
 #[tokio::main]
 async fn main() -> Result<()> {
-    // Generates a wallet.log file with logs for debugging.
-    let logger_output_config = fern_logger::LoggerOutputConfigBuilder::new()
-        .name("wallet.log")
-        .target_exclusions(&["h2", "hyper", "rustls"])
-        .level_filter(log::LevelFilter::Debug);
-    let config = fern_logger::LoggerConfig::build()
-        .with_output(logger_output_config)
-        .finish();
-    fern_logger::logger_init(config).unwrap();
-
     // This example uses secrets in environment variables for simplicity which should not be done in production.
     dotenvy::dotenv().ok();
 
-    let client_options = ClientOptions::new()
-        .with_node(&std::env::var("NODE_URL").unwrap())?
-        .with_ignore_node_health();
-
-    let secret_manager =
-        MnemonicSecretManager::try_from_mnemonic(&std::env::var("NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1").unwrap())?;
-
     let wallet = Wallet::builder()
-        .with_secret_manager(SecretManager::Mnemonic(secret_manager))
-        .with_client_options(client_options)
-        .with_coin_type(SHIMMER_COIN_TYPE)
+        // TODO: is this needed?
```

Comment:
Is it?

Reply:
hmm, it works without, which is why I put this there as reminder. If it's not required for the example I would remove it to keep the example as simple as possible. I think it's probably generally not a good idea to use a node that's not healthy for participating in an event, but @Thoralf-M or @thibault-martinez probably know better.

Reply:
Not needed, you can remove